internal/pkg/template/templates/workloads/services/static-site/cf.yml

```diff
@@ -17,15 +17,271 @@ Parameters:
+                SourceBucket: event.ResourceProperties.SourceBucket,
+                AssetMappingFilePath: event.ResourceProperties.AssetMappingFilePath,
+                DestinationBucket: event.ResourceProperties.DestinationBucket,
+              })
+            };
+            sf.startSyncExecution(params, (err, data) => {
+              if (err) {
+                console.log("error starting step function", err)
+                response.send(event, context, response.FAILED, {"error": err.message}, event.PhysicalResourceId)
+                return;
+              }
+
+              console.log("finished step function execution:", data)
+
+              if (data.status === "FAILED") {
+                response.send(event, context, response.FAILED, {"error": data.cause}, event.PhysicalResourceId)
+                return;
+              }
+
+              response.send(event, context, response.SUCCESS, {}, event.PhysicalResourceId)
+            });
+          };
+
+  TriggerStateMachineFunctionRole:
+    Metadata:
+      aws:copilot:description: An IAM Role for the lambda that triggers the file copying state machine.
+    Type: AWS::IAM::Role
+    Properties:
+      {{- if .PermissionsBoundary}}
+      PermissionsBoundary: !Sub arn:${AWS::Partition}:iam::${AWS::AccountId}:policy/{{.PermissionsBoundary}}
+      {{- end}}
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action: sts:AssumeRole
+      Policies:
+        - PolicyName: TriggerStateMachine
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action: states:StartSyncExecution
+                Resource: !GetAtt CopyAssetsStateMachine.Arn
+      ManagedPolicyArns:
+        - !Sub arn:${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+
+  TriggerStateMachineAction:
+    Metadata:
+      aws:copilot:description: A custom resource to trigger the file copying state machine
+    Type: Custom::TriggerStateMachine
+    DependsOn: CopyAssetsStateMachine
+    Properties:
+      ServiceToken: !GetAtt TriggerStateMachineFunction.Arn
+      StateMachineARN: !GetAtt CopyAssetsStateMachine.Arn
+      SourceBucket: {{ .AssetMappingFileBucket }}
+      AssetMappingFilePath: {{ .AssetMappingFilePath }}
+      DestinationBucket: !Ref Bucket
+
+  CopyAssetsStateMachine:
+    Type: AWS::StepFunctions::StateMachine
+    Properties:
+      RoleArn: !GetAtt CopyAssetsStateMachineRole.Arn
+      StateMachineType: EXPRESS
+      Definition:
+        Comment: A test state machine
+        StartAt: GetMappingFile
+        States:
+          GetMappingFile:
+            Type: Task
+            Parameters:
+              Bucket.$: $.SourceBucket
+              Key.$: $.AssetMappingFilePath
+            Resource: arn:aws:states:::aws-sdk:s3:getObject
```

Comment:
Awesome! I was thinking using the trigger lambda to get and parse the mapping file but this looks nicer